AnkiHubSoftware/ankihub_addon
# [BUILD-748] feat: Add function to send the daily summary review data after the sync

## Related issues
- [BUILD-748](https://ankihub.atlassian.net/browse/BUILD-748)


## Proposed changes
This PR makes these changes:
* Add the function `send_daily_review_summaries` which will send to the AnkiHub server the compiled data of the daily reviews
* Call this function in the background after the user sync
* Update the `feature_flags.py` module and its use to avoid the need to update the dataclass if a new feature flag is created on the server


## How to reproduce
1. Run the instance of the ankihub server and make sure that the feature flag `daily_card_review_summary` is enabled
2. Open a deck and review a card
3. Open the anki database and update the ID of the newly created `revlog` row to the value of a timestamp of yesterday's date
4.  Click on the "Sync with AnkiHub" option on the Ankihub menu
5. Go to a shell instance of the Ankihub server and check if a register of the `DailyCardReviewSummary` was created successfully


[BUILD-748]: https://ankihub.atlassian.net/browse/BUILD-748?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
